x/merkledb/proof_test.go

```diff
@@ -1532,3 +1456,125 @@ func TestVerifyProofPath(t *testing.T) {
 		})
 	}
 }
+
```

Comment:
I think this change introduces a number of panic vulns. ex:
```
func TestProofNodeUnmarshalProtoNilMaybeBytes(t *testing.T) {
	rand := rand.New(rand.NewSource(1337)) // #nosec G404

	node := newRandomProofNode(rand)
	protoNode := node.ToProto()

	protoNode.ValueOrHash = nil

	protoNodeBytes, err := proto.Marshal(protoNode)
	require.NoError(t, err)

	var unmarshalledProtoNode syncpb.ProofNode
	err = proto.Unmarshal(protoNodeBytes, &unmarshalledProtoNode)
	require.NoError(t, err)

	var unmarshaledNode ProofNode
	err = unmarshaledNode.UnmarshalProto(protoNode) // Panics
	require.NoError(t, err)
}
```
I believe there are similar concerns with the Range Proofs

Reply:
I think this is adequately handled now